scripts/ubuntu_dev_env_setup.sh

```diff
@@ -1,74 +1,321 @@
 # kubectl
+install_kubectl() {
+    echo -e "${YELLOW}Installing Kubectl...${RESET}"
+    curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+    sudo install -o root -g root -m 0755 kubectl /usr/local/bin/kubectl
+    # Verify kubectl has installed
+    kubectl version --client || echo -e "${RED}kubectl installation verification failed!${RESET}"
+    # Remove installation "junk"
+    rm -f kubectl
+
+}
 
 # K9s
+install_k9s() {
+    echo -e "${YELLOW}Installing K9s...${RESET}"
+    # Find the latest K9s Debian package URL from GitHub releases
+    latest_k9s=$(curl -s https://api.github.com/repos/derailed/k9s/releases/latest | grep "browser_download_url.*amd64.deb" | cut -d '"' -f 4)
+    # Download the latest K9s .deb package
+    wget "$latest_k9s"
+    # Install K9s using the .deb package
+    sudo apt install ./"$(basename "$latest_k9s")"
+    # Verify K9s installation
+    k9s version || echo -e "${RED}K9s installation verification failed!${RESET}"
+    # Remove Installation "junk"
+    rm -f "$(basename "$latest_k9s")"
+
+}
 
 # minikube
+install_minikube() {
+    echo -e "${YELLOW}Installing Minikube...${RESET}"
+    curl -LO https://github.com/kubernetes/minikube/releases/latest/download/minikube-linux-amd64
+    sudo install minikube-linux-amd64 /usr/local/bin/minikube
+    # Verify minikube has installed
+    minikube version || echo -e "${RED}minikube installation verification failed!${RESET}"
+    # Remove installation "junk"
+    rm -f minikube-linux-amd64
+
+}
 
 # Oh My Posh
+install_oh_my_posh() {
+    echo -e "${YELLOW}Installing Oh My Posh...${RESET}"
+    curl -s https://ohmyposh.dev/install.sh | bash -s
+    # Verify Oh My Posh has installed
+    oh-my-posh --version || echo -e "${RED}Oh My Posh installation verification failed!${RESET}"
+
+}
 
 # VScode extensions
 # ====================================
-# terraform, kubernetes tools, azure tools, bicep, gitlen, spell checker
+# terraform, kubernetes tools, azure tools, bicep, gitlen, spell checker, helm
+install_vscode_extensions() {
+    echo -e "${YELLOW}Installing VS Code extensions...${RESET}"
+    code --install-extension hashicorp.terraform
+    code --install-extension ms-kubernetes-tools.vscode-kubernetes-tools
+    code --install-extension msazurermtools.azurerm-vscode-tools
+    code --install-extension bicep.bicep
+    code --install-extension eamodio.gitlens
+    code --install-extension streetsidesoftware.code-spell-checker
+}
 
 # Config git with ssh authentication
+# The following portion would require manual inputs"
+configure_git() {
+    echo -e "${YELLOW}Configuring Git with SSH...${RESET}"
+    echo "Enter your primary email for git:"
+    read git_email
+    git config --global user.email "$git_email"
+		
+    echo "Enter your name for git:"
+    read git_name
+    git config --global user.name "$git_name"
+    
+    echo "Generating SSH key for GitHub..."
```

Comment:
I think we should add a check to see if the id_rsa file exists. Otherwise, it would overwrite the existing one.